moveit_core/collision_detection_fcl/src/collision_common.cpp

```diff
@@ -404,29 +405,33 @@ bool distanceCallback(fcl::CollisionObject* o1, fcl::CollisionObject* o2, void*
             cd2->ptr.link :
             (cd2->type == BodyTypes::ROBOT_ATTACHED ? cd2->ptr.ab->getAttachedLink() : nullptr);
 
+    if (!l1 || cdata->req->active_components_only->find(l1) == cdata->req->active_components_only->end())
+      active1 = false;
+
+    if (!l2 || cdata->req->active_components_only->find(l2) == cdata->req->active_components_only->end())
+      active2 = false;
+
     // If neither of the involved components is active
-    if ((!l1 || cdata->active_components_only_->find(l1) == cdata->active_components_only_->end()) &&
-        (!l2 || cdata->active_components_only_->find(l2) == cdata->active_components_only_->end()))
+    if (active1 == false && active2 == false)
     {
-      min_dist = cdata->res_->distance;
-      return cdata->done_;
+      return false;
     }
   }
 
   // use the collision matrix (if any) to avoid certain distance checks
   bool always_allow_collision = false;
-  if (cdata->acm_)
+  if (cdata->req->acm)
   {
     AllowedCollision::Type type;
 
-    bool found = cdata->acm_->getAllowedCollision(cd1->getID(), cd2->getID(), type);
+    bool found = cdata->req->acm->getAllowedCollision(cd1->getID(), cd2->getID(), type);
     if (found)
     {
       // if we have an entry in the collision matrix, we read it
       if (type == AllowedCollision::ALWAYS)
       {
         always_allow_collision = true;
-        if (cdata->req_->verbose)
+        if (!cdata->req->verbose)
```

Comment:
Why did you reversed the meaning of `verbose` everywhere in this code???

Reply:
This was done in error. I will correct it.